include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -0,0 +1,1018 @@
+    SizeType i_subm_el = distr.globalTileElementDistance<Coord::Row>(i_begin, i_tile);
+    for (SizeType j_tile = i_begin; j_tile <= i_end; ++j_tile) {
+      SizeType j_subm_el = distr.globalTileElementDistance<Coord::Col>(i_begin, j_tile);
+      auto sum_first_tile_cols_fn = [i_subm_el, j_subm_el](SizeType k, const ConstTileType& ws_tile,
+                                                           const TileType& ws_row_tile) {
+        if (i_subm_el >= k || j_subm_el >= k)
+          return;
+
+        SizeType ncols = std::min(k - j_subm_el, ws_tile.size().cols());
+        for (SizeType j = 0; j < ncols; ++j) {
+          ws_row_tile(TileElementIndex(0, j)) += ws_tile(TileElementIndex(0, j));
+        }
+      };
+
+      ex::when_all(ex::keep_future(k_fut), ws.read_sender(GlobalTileIndex(i_tile, j_tile)),
+                   ws.readwrite_sender(GlobalTileIndex(i_begin, j_tile))) |
+          di::transform(di::Policy<Backend::MC>(), std::move(sum_first_tile_cols_fn)) |
+          ex::ensure_started();
+    }
+  }
+
+  // Normalize column vectors
+  for (SizeType i_tile = i_begin; i_tile <= i_end; ++i_tile) {
+    SizeType i_subm_el = distr.globalTileElementDistance<Coord::Row>(i_begin, i_tile);
+    for (SizeType j_tile = i_begin; j_tile <= i_end; ++j_tile) {
+      SizeType j_subm_el = distr.globalTileElementDistance<Coord::Col>(i_begin, j_tile);
+      auto scale_fn = [i_subm_el, j_subm_el](SizeType k, const ConstTileType& ws_tile,
+                                             const TileType& evecs_tile) {
+        if (i_subm_el >= k || j_subm_el >= k)
+          return;
+
+        SizeType nrows = std::min(k - i_subm_el, evecs_tile.size().rows());
+        SizeType ncols = std::min(k - j_subm_el, evecs_tile.size().cols());
+
+        for (SizeType j = 0; j < ncols; ++j) {
+          for (SizeType i = 0; i < nrows; ++i) {
+            T& evecs_el = evecs_tile(TileElementIndex(i, j));
+            evecs_el = evecs_el / std::sqrt(ws_tile(TileElementIndex(0, j)));
+          }
+        }
+      };
+
+      ex::when_all(ex::keep_future(k_fut), ws.read_sender(GlobalTileIndex(i_begin, j_tile)),
+                   evecs.readwrite_sender(GlobalTileIndex(i_tile, j_tile))) |
+          di::transform(di::Policy<Backend::MC>(), std::move(scale_fn)) | ex::ensure_started();
+    }
+  }
+}
+
+template <class T, Device Source, Device Destination>
+void copySubMatrix(SizeType i_begin, SizeType i_end, Matrix<const T, Source>& source,
+                   Matrix<T, Destination>& dest) {
+  namespace ex = pika::execution::experimental;
+
+  for (SizeType j = i_begin; j <= i_end; ++j) {
+    for (SizeType i = i_begin; i <= i_end; ++i) {
+      ex::when_all(source.read_sender(LocalTileIndex(i, j)),
+                   dest.readwrite_sender(LocalTileIndex(i, j))) |
+          matrix::copy(dlaf::internal::Policy<matrix::internal::CopyBackend_v<Source, Destination>>{}) |
+          ex::start_detached();
```

Comment:
```suggestion
      ex::start_detached(ex::when_all(source.read_sender(LocalTileIndex(i, j)),
                   dest.readwrite_sender(LocalTileIndex(i, j))) |
          matrix::copy(dlaf::internal::Policy<matrix::internal::CopyBackend_v<Source, Destination>>{}));
```
because the reference implementation does not allow piping into `start_detached`.